### DISCUSSION TITLE
模仿 Safari 阅读模式样式

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1595] [插件说明]
描述
---

> 提供类似 **Safari 阅读模式** 的主题

地址
---

[https://simpread.ksria.cn/plugins/details/tbjTx33iAA](https://simpread.ksria.cn/plugins/details/tbjTx33iAA)

截图
---

[![NTRbHP.md.png](https://s1.ax1x.com/2020/07/01/NTRbHP.md.png)](https://imgchr.com/i/NTRbHP)

[![NTR7nI.md.png](https://s1.ax1x.com/2020/07/01/NTR7nI.md.png)](https://imgchr.com/i/NTR7nI)

[![NTRLAf.md.png](https://s1.ax1x.com/2020/07/01/NTRLAf.md.png)](https://imgchr.com/i/NTRLAf)

[![NTRHBt.md.png](https://s1.ax1x.com/2020/07/01/NTRHBt.md.png)](https://imgchr.com/i/NTRHBt)

功能
---

1. 跟随 **主题变化**

2. 类似 Safari 阅读模式 **点击空白处** 自动退出阅读模式（可更改）

3. 目录居中对齐。（ 1.0.6 版新增功能）

选项
---

![image](https://user-images.githubusercontent.com/81074/124222324-1d392a00-db34-11eb-82e7-f6d5aace9133.png)

![image](https://user-images.githubusercontent.com/81074/124221619-b36c5080-db32-11eb-8ff5-5d69173a264c.png)

暗色模式
---

> 如果你在系统的暗色模式下出现样式错误，请**手动设置阅读模式的主题为暗色模式，即可解决**，细节[请看这里](https://github.com/Kenshin/simpread/discussions/1595#discussioncomment-1231214)。

![image](https://user-images.githubusercontent.com/81074/130737732-4632cd28-5e25-468f-8e40-5c37a7d90950.png)

更新日志
---

### 1.1.0

- `Fix` 当浏览器尺寸过小时 `目录宽度` 计算错误

- `Add` 版本检查（选项卡左下角版本号，点击可验证是否有新版）

- `Update` 选项卡升级到 2.0

### 1.0.9

- `Optimize` 优化算法，让此插件加载更快。

### 1.0.8

- `Fix` 系统暗色模式、选项页 → 高级设定 → （关闭）随系统暗色模式，导致主题色错误的问题 [#5813](https://github.com/Kenshin/simpread/issues/5813)

### 1.0.7

- `Fix` 暗色模式下样式错误

- `Add` 跟随系统的暗色模式

### 1.0.6

> 注意从 **0.0.5 → 1.0.6** 是因为插件的版本号改为以 `1.0.0` 为基准

- `Optimize` 左侧目录居中对齐

### 0.0.5

- `Fix` `window.addEventListener( 'simpread-plugin' )` 错误

-  `Rework` Toc 在 Safari 下的样式逻辑

### 0.0.4

- `Add` 选项，包括：点击空白退出 or 不退出。 
    https://github.com/Kenshin/simpread/issues/2414 https://github.com/Kenshin/simpread/issues/2292

### 0.0.3( 2020-07-01 )

- `Fix` 不与主题适配的错误 [#742](https://github.com/Kenshin/simpread/issues/742) [#609](https://github.com/Kenshin/simpread/issues/609)

- `Add` 点击空白退出阅读模式的功能 [#753](https://github.com/Kenshin/simpread/issues/753)

**Comment by LXY631** [comment]
模仿safari插件 直接调用效果
![111](https://user-images.githubusercontent.com/42035547/126477930-18fd43a8-c605-4655-a87b-303735389174.png)
点击样式 任意一种主题色 后效果
![222](https://user-images.githubusercontent.com/42035547/126478177-bdf3f684-3e60-4e83-848d-feebde91810a.png)
目前估计是高级设定-暗色模式的干扰，关闭后正常，望修复。

**Comment by OptimusPrimeme** [comment]
好像无法跟随暗色模式的设置自动切换，导致浏览器切换为深色模式时，效果如下，望修复
![image](https://user-images.githubusercontent.com/22427706/130629933-10666180-fbf1-4304-b285-6dec6d6ac98e.png)

**Reply by Kenshin** [reply]
刷新页面也不好使吗？

**Reply by OptimusPrimeme** [reply]
> 刷新页面也不好使吗？

嗯，[这是文章的连接](https://sspai.com/post/68441)
这个插件实现的样式挺舒服的，不知道能不能通过自定义样式实现

**Reply by Kenshin** [reply]
嗨，我知道是什么问题了。


### 暗色模式下 Safari 的问题

当你使用了系统的暗色模式，且使用了简悦的暗色模式（如下图

![image](https://user-images.githubusercontent.com/81074/130737500-1845e336-2b43-4f90-af51-df3645d9a296.png)

并且阅读模式的主题却使用了亮色模式

![image](https://user-images.githubusercontent.com/81074/130737624-9ace6531-bde5-4e47-a721-4e2ba320ea1c.png)

就会存在此问题。

### 解决方案

> 手动设置阅读模式的主题为暗色模式即可解决。

![image](https://user-images.githubusercontent.com/81074/130737732-4632cd28-5e25-468f-8e40-5c37a7d90950.png)

**Reply by OptimusPrimeme** [reply]
嗯确实解决了问题，谢谢啦🌹，不过能不能自动切换？毕竟原来不用这个插件时，阅读模式的主题就算使用了亮色，也可以通过暗色设置自动切换

**Reply by Kenshin** [reply]
> 嗯确实解决了问题，谢谢啦🌹，不过能不能自动切换？毕竟原来不用这个插件时，阅读模式的主题就算使用了亮色，也可以通过暗色设置自动切换

看看能否下个版本搞定自动切换的功能。

**Comment by yiepase** [comment]
![20211022_081110](https://user-images.githubusercontent.com/78405306/138373895-29449aaa-55a0-4026-838b-ad6b8341ab26.png)
你好目录出问题了 无法隐藏

**Reply by Kenshin** [reply]
嗨，请提供出现问题的 URL

**Reply by yiepase** [reply]
https://zhuanlan.zhihu.com/p/322673517
插件界面发现字体等失去颜色

**Reply by yiepase** [reply]
有时

**Reply by Kenshin** [reply]
> https://zhuanlan.zhihu.com/p/322673517 插件界面发现字体等失去颜色

失去颜色是指？（最好接图

另外，上面提到的问题，修复后会通知你。
